server/src/test/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequestTests.java

```diff
@@ -31,21 +31,137 @@
-        assertEquals("Force-merge indices [], maxSegments[-1], onlyExpungeDeletes[false], flush[true]", request.getDescription());
+        assertEquals(
+            "Force-merge indices [], maxSegments[-1], onlyExpungeDeletes[false], flush[true], primaryOnly[false]",
+            request.getDescription()
+        );
 
         request = new ForceMergeRequest("shop", "blog");
-        assertEquals("Force-merge indices [shop, blog], maxSegments[-1], onlyExpungeDeletes[false], flush[true]", request.getDescription());
+        assertEquals(
+            "Force-merge indices [shop, blog], maxSegments[-1], onlyExpungeDeletes[false], flush[true], primaryOnly[false]",
+            request.getDescription()
+        );
 
         request = new ForceMergeRequest();
         request.maxNumSegments(12);
         request.onlyExpungeDeletes(true);
         request.flush(false);
-        assertEquals("Force-merge indices [], maxSegments[12], onlyExpungeDeletes[true], flush[false]", request.getDescription());
+        request.primaryOnly(true);
+        assertEquals(
+            "Force-merge indices [], maxSegments[12], onlyExpungeDeletes[true], flush[false], primaryOnly[true]",
+            request.getDescription()
+        );
+    }
+
+    public void testToString() {
+        ForceMergeRequest request = new ForceMergeRequest();
+        assertEquals("ForceMergeRequest{maxNumSegments=-1, onlyExpungeDeletes=false, flush=true, primaryOnly=false}", request.toString());
+
+        request = new ForceMergeRequest();
+        request.maxNumSegments(12);
+        request.onlyExpungeDeletes(true);
+        request.flush(false);
+        request.primaryOnly(true);
+        assertEquals("ForceMergeRequest{maxNumSegments=12, onlyExpungeDeletes=true, flush=false, primaryOnly=true}", request.toString());
+    }
+
+    public void testSerialization() throws Exception {
+        final ForceMergeRequest request = randomRequest();
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            request.writeTo(out);
+
+            final ForceMergeRequest deserializedRequest;
+            try (StreamInput in = out.bytes().streamInput()) {
+                deserializedRequest = new ForceMergeRequest(in);
+            }
+            assertEquals(request.maxNumSegments(), deserializedRequest.maxNumSegments());
+            assertEquals(request.onlyExpungeDeletes(), deserializedRequest.onlyExpungeDeletes());
+            assertEquals(request.flush(), deserializedRequest.flush());
+            assertEquals(request.primaryOnly(), deserializedRequest.primaryOnly());
+            assertEquals(request.forceMergeUUID(), deserializedRequest.forceMergeUUID());
+        }
+    }
+
+    public void testBwcSerialization() throws Exception {
+        {
+            final ForceMergeRequest sample = randomRequest();
+            final Version compatibleVersion = VersionUtils.randomCompatibleVersion(random(), Version.CURRENT);
```

Comment:
if i am not wrong, for 3.0.0 the compatible version from 2.x would be 2.12.0 right? can we directly use 2.11.0 and test if it works fine with it?

Reply:
Should we test 2.11.0 in 2.x branch? After this PR has been merged and backported into 2.x, then 2.11.0 will be tested in 2.x branch. In main branch, we may only test 3.0.0 and its compatibleVersion?

Reply:
keeping the test in 2.x is also fine with me. but lets atleast test it manually with 2.11 to make sure nothing is breaking before merging the changes.

Reply:
3.0.0 is not compatible with 2.11, nodes with those two versions cannot join as a cluster, so we can test 2.11 in 2.x branch.